carta/cpp/CartaLib/AxisDisplayInfo.h

```diff
@@ -0,0 +1,134 @@
+/***
+ * Information about how an axis should be displayed.
+ */
+
```

Comment:
The purpose of this class is not clear to me from the comment. By glancing at the code, it seems to be used in places where axes need to be permuted in an ImageInterface...

For axis permutations I was thinking of implementing something generic, e.g. by adding a method to ImageInterface:

virtual Image::ImageInterface * permuted(
    const std::vector<int> & indices)

which would return a special instance of ImageInterface. This way you could permute axes and/or even remove some, without having to insert permutation specific code anywhere else...